src/shogun/distance/EuclideanDistance.h

```diff
@@ -125,6 +161,21 @@ class CEuclideanDistance: public CRealDistance
 	protected:
 		/** if application of sqrt on matrix computation is disabled */
 		bool disable_sqrt;
+
+		/** if squared norms of rhs features are used */
+		bool m_rhs_sq_norms;
+
+		/** if squared norms of lhs features are used */
+		bool m_lhs_sq_norms;
+
+		/** i dot products are used in distance computations */
+		bool m_dot_enabled;
+
+		/** squared norms from features of right hand side to be used if dot enabled */
+		SGVector<float64_t> m_rhs_squared_norms;
```

Comment:
What about:
storing the lhs norms always
not storing the rhs norms

Offer a method that computes all pairwise distance between lhs and rhs, and precompute the rhs norms in there?

H